Sources/NIOCore/EventLoop.swift

```diff
@@ -356,6 +356,25 @@ public protocol EventLoop: EventLoopGroup {
     /// It is valid for an `EventLoop` not to implement any of the two `_promise` functions. If either of them are implemented,
     /// however, both of them should be implemented.
     func _promiseCompleted(futureIdentifier: _NIOEventLoopFutureIdentifier) -> (file: StaticString, line: UInt)?
+
+    /// Schedule a callback at a given time.
+    ///
+    /// - NOTE: Event loops that provide a custom scheduled callback implementation **must** also implement
+    ///         `cancelScheduledCallback`. Failure to do so will result in a runtime error.
+    @discardableResult
+    func scheduleCallback(at deadline: NIODeadline, handler: some NIOScheduledCallbackHandler) throws -> NIOScheduledCallback
```

Comment:
NIT: I personally find `callback` an overloaded term here since it is mostly used for meaning `closure` in Swift/ObjC. I liked the previous timer naming.

Reply:
> I liked the previous timer naming.
😭 😰 

IMO @glbrntt is right here. This _is_ a callback in the truest sense of what is being _scheduled_.

Reply:
Maybe. It was just a nit and I am fine with both